src/types.ts

```diff
@@ -63,35 +39,37 @@ export type ResourceRelationship = {
   id: string;
 };
 
-export type Metadata = {
-  [key: string]: string | number | boolean | Metadata;
+export type Meta = {
+  [key: string]: string | number | boolean | Meta;
 };
 
 export type JsonApiDocument<
   ResourceT = Resource,
   RelatedResourcesT = Resource
 > = {
   data: ResourceT | ResourceT[];
-  errors?: JsonApiError[];
-  meta?: Metadata;
+  meta?: Meta;
   included?: RelatedResourcesT[];
 };
 
-export type JsonApiErrorDocument<ErrorCodeT = ErrorCode> = {
-  errors?: JsonApiError<ErrorCodeT>[];
-  meta?: Metadata;
-};
-
-export type JsonApiRequest = {
-  data?: Resource;
-  meta?: Metadata;
+export type JsonApiErrorsDocument = {
+  errors?: JsonApiError[];
+  meta?: Meta;
 };
 
-export type JsonApiError<ErrorCodeT = ErrorCode> = {
-  status: HttpStatusCode;
-  code: ErrorCodeT;
-  title: string;
-  detail: string;
+export type JsonApiError = {
+  id?: string;
+  status?: HttpStatusCode;
```

Comment:
@Exelord Both `status` and `code` should be mandatory.

Reply:
> An error object MAY have the following members:
...
status: the HTTP status code applicable to this problem, expressed as a string value.
code: an application-specific error code, expressed as a string value.
...

Why should we require them?

Reply:
Beacuse otherwise there's no point on sending an error response if you don't have at very least a status and code.